lib/Models/CkanCatalogGroup.ts

```diff
@@ -0,0 +1,432 @@
+import i18next from "i18next";
+import LoadableStratum from "./LoadableStratum";
+import { computed, runInAction, action } from "mobx";
+import { BaseModel } from "./Model";
+import UrlMixin from "../ModelMixins/UrlMixin";
+import GroupMixin from "../ModelMixins/GroupMixin";
+import CatalogMemberMixin from "../ModelMixins/CatalogMemberMixin";
+import CreateModel from "./CreateModel";
+import CatalogGroup from "./CatalogGroupNew";
+import StratumOrder from "./StratumOrder";
+import URI from "urijs";
+import proxyCatalogItemUrl from "./proxyCatalogItemUrl";
+import TerriaError from "../Core/TerriaError";
+import loadJson from "../Core/loadJson";
+import isDefined from "../Core/isDefined";
+import createStratumInstance from "./createStratumInstance";
+import { InfoSectionTraits } from "../Traits/CatalogMemberTraits";
+import CkanItemReference from "./CkanItemReference";
+import filterOutUndefined from "../Core/filterOutUndefined";
+import ModelReference from "../Traits/ModelReference";
+import CkanCatalogGroupTraits from "../Traits/CkanCatalogGroupTraits";
+import CommonStrata from "./CommonStrata";
+import Terria from "./Terria";
+import replaceUnderscores from "../Core/replaceUnderscores";
+import {
+  CkanDataset,
+  CkanResource,
+  CkanSearchResult,
+  CkanServerResponse
+} from "./CkanDefinitions";
+
+export class CkanServerStratum extends LoadableStratum(CkanCatalogGroupTraits) {
+  static stratumName = "ckanServer";
+  groups: CatalogGroup[] = [];
+  filteredGroups: CatalogGroup[] = [];
+  datasets: CkanDataset[] = [];
+  filteredDatasets: CkanDataset[] = [];
+
+  constructor(
+    readonly _catalogGroup: CkanCatalogGroup,
+    private readonly _ckanResponse: CkanServerResponse
+  ) {
+    super();
+    this.datasets = this.setDatasets();
+    this.filteredDatasets = this.setFilteredDatasets();
+    this.groups = this.setGroups();
+    this.filteredGroups = this.setFilteredGroups();
+  }
+
+  duplicateLoadableStratum(model: BaseModel): this {
+    return new CkanServerStratum(
+      model as CkanCatalogGroup,
+      this._ckanResponse
+    ) as this;
+  }
+
+  static async load(
+    catalogGroup: CkanCatalogGroup
+  ): Promise<CkanServerStratum | undefined> {
+    var terria = catalogGroup.terria;
+
+    let ckanServerResponse: CkanServerResponse | undefined = undefined;
+
+    // Each item in the array causes an independent request to the CKAN, and the results are concatenated
+    for (var i = 0; i < catalogGroup.filterQuery.length; ++i) {
+      const filterQuery = catalogGroup.filterQuery[i];
+      var uri = new URI(catalogGroup.url)
+        .segment("api/3/action/package_search")
+        .addQuery({ start: 0, rows: 1000, sort: "metadata_created asc" });
+
+      Object.keys(filterQuery).forEach((key: string) =>
+        // @ts-ignore
```

Comment:
My TS skills aren't good enough to get rid of the use of this `@ts-ignore`

Reply:
Hah. I worked out what it was. See other comment

Reply:
In future I think it's better to use `<SomeType>` or `<any>` (or the `x as SomeType` form) rather than `ts-ignore`. I think ignoring a whole line is almost always going to ignore too much, and it also has the problem that prettier could place more or less code on that line in the future, which could then either fail type checking or mask other type errors even if there's little chance of `ts-ignore` masking errors at first.